src/components/withTracker.jsx

```diff
@@ -0,0 +1,44 @@
+import React, { Component } from 'react'
+import GoogleAnalytics from 'react-ga'
+
+if (process.env.REACT_APP_GA_TRACKING_ID) {
+  console.log('Initializing Google Analytics with tracking ID: ', process.env.REACT_APP_GA_TRACKING_ID)
+  GoogleAnalytics.initialize(process.env.REACT_APP_GA_TRACKING_ID)
+}
+
+export const withTracker = (WrappedComponent, options = {}) => {
+  const trackPage = page => {
+    if (process.env.REACT_APP_GA_TRACKING_ID) {
+      console.log('Tracking: ', page)
+      GoogleAnalytics.set({
+        page,
+        ...options
+      })
+      GoogleAnalytics.pageview(page)
```

Comment:
This is such a different way of using GA then I've ever seen before. What are the benefits to doing this instead of just including the script tag in the html file?

Reply:
This is the recommended way of tracking a React Router single page app